### PR TITLE
added support for athrow and dup_x1 op codes and ginstance, only for object types

### PR DESCRIPTION
**Pull request by joseph-isaacs**

### CODE REVIEW

File: src/Codec/JVM/ASM/Code.hs [hunk 1]
```diff
@@ -283,6 +291,17 @@ gwide opcode n = wideInstr
                      <> IT.op opcode
                      <> IT.bytes (packI16 $ fromIntegral n)
 
+ginstanceof :: FieldType -> Code
+ginstanceof ft@(ObjectType (IClassName className ))  = mkCode cs $
+  IT.withOffset $ \offset -> fold
```
Comment (rahulmutt) [inline src/Codec/JVM/ASM/Code.hs]: You don't use `offset` anywhere - is this intended?

**Review by joseph-isaacs — COMMENTED** [review]
Change the ginstance to use mkCode

File: src/Codec/JVM/ASM/Code.hs [hunk 2]
```diff
@@ -653,6 +673,16 @@ swap ft1 ft2 =
...
+dup_x1 :: FieldType -> FieldType -> Code
+dup_x1 ft1 ft2 =
+  mkCode' $
+    IT.op OP.dup_x1
+  <> modifyStack ( CF.push ft2
+                 . CF.push ft1
+                 . CF.push ft2
+                 . CF.pop  ft2
+                 . CF.pop  ft1)
+
```
Comment (rahulmutt) [inline src/Codec/JVM/ASM/Code.hs]: Swap the ordering of `pop ft2` and `pop ft1` and I'll merge.

**Review by rahulmutt — CHANGES_REQUESTED** [review]

File: src/Codec/JVM/ASM/Code.hs (hunk 2 above)
Comment (joseph-isaacs) [reply src/Codec/JVM/ASM/Code.hs]: Fixed

**Comment by rahulmutt** [comment]
It's hard for me to see what changes you did. Can you close this pull request and open a new one (after  rebasing your fork)?

**Pull request closed without merging**